crates/network-scanner/src/task_utils.rs

```diff
@@ -25,7 +24,8 @@ pub(crate) struct TaskExecutionContext {
     pub port_sender: PortSender,
     pub port_receiver: Arc<Mutex<PortReceiver>>,
 
-    pub ip_cache: Arc<DashMap<IpAddr, Option<String>>>,
+    pub ip_cache: Arc<parking_lot::Mutex<HashMap<IpAddr, Option<String>>>>,
```

Comment:
suggestion: You could use https://docs.rs/parking_lot/latest/parking_lot/type.RwLock.html (if we have more reads that writes)

Reply:
updated